packages/docusaurus/src/webpack/templates/ssr.html.template.ts

```diff
@@ -16,20 +16,18 @@ export default `
     <% }); %>
     <%~ it.headTags %>
     <% it.stylesheets.forEach((stylesheet) => { %>
+      <link rel="preload" href="<%= it.baseUrl %><%= stylesheet %>" as="style" />
       <link rel="stylesheet" href="<%= it.baseUrl %><%= stylesheet %>" />
```

Comment:
I'm not sure to understand how it applies here:

<img width="686" alt="CleanShot 2022-09-28 at 13 16 54@2x" src="https://user-images.githubusercontent.com/749374/192765768-40f976ca-5182-432f-bdd7-1f10aa0e511e.png">


The stylesheet is not really discovered late, but quite eagerly in the document head. How can preloading something that is already loading changes anything?

> Similar to inlining code in a <script> tag, inline critical styles required for the first paint inside a <style> block at the head of the HTML page. Then load the rest of the styles asynchronously using the preload link (see [Defer unused CSS](https://web.dev/defer-non-critical-css)).

We don't have a good way to separate critical CSS from non-critical CSS in Docusaurus for now. All our CSS is in a critical CSS file and should always load eagerly

Reply:
Do you have a ref for this change? How does it improve anything?

Reply:
> Do you have a ref for this change? How does it improve anything?

Sorry, actually it needed to be moved above the meta tags so browser can make request to relevant CSS file before hand which can improve performance of the loading speed. Since CSS files are placed really bottom of head tag which might take time to html parser to read it.

Extremely sorry I will update it ASAP :(

Related PR in next js -: https://github.com/vercel/next.js/pull/9486
For more ref -: https://web.dev/preload-critical-assets/

Reply:
> Do you have a ref for this change? How does it improve anything?

@slorber  Done 🤝